src/app/libp2p_helper/src/codanet.go

```diff
@@ -118,9 +137,51 @@ func (cm *CodaConnectionManager) ClosedStream(net network.Network, stream networ
 	cm.p2pManager.Notifee().ClosedStream(net, stream)
 }
 func (cm *CodaConnectionManager) Connected(net network.Network, c network.Conn) {
+	logger.Debugf("%s connected to %s", c.LocalPeer(), c.RemotePeer())
 	cm.OnConnect(net, c)
 	cm.p2pManager.Notifee().Connected(net, c)
+
+	if len(net.Peers()) <= cm.maxConnections {
```

Comment:
You can call  `info := cm.p2pManager.GetInfo()` to get an object with properties `ConnCount`, `HighWater` and `LowWater` and then check `info.ConnCount <= info.HighWater`